pkg/cluster/fixssh.go

```diff
@@ -82,22 +112,34 @@ func (m *manager) fixSSH(ctx context.Context) error {
 			}
 		}
 	}
+	return nil
+}
 
+func (m *manager) removeBackendPoolsFromNIC(ctx context.Context, resourceGroup, nicName string, nic *mgmtnetwork.Interface) error {
+	ipc := (*nic.InterfacePropertiesFormat.IPConfigurations)[0]
+	if ipc.LoadBalancerBackendAddressPools != nil {
+		m.log.Printf("Removing Loadbalancer Backend Address Pools from NIC %s with no VMs attached", nicName)
+		*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = []mgmtnetwork.BackendAddressPool{}
+		return m.interfaces.CreateOrUpdateAndWait(ctx, resourceGroup, nicName, *nic)
+	}
 	return nil
 }
 
-func updateNIC(nic *mgmtnetwork.Interface, lb *mgmtnetwork.LoadBalancer, i int) bool {
+func updateNIC(nic *mgmtnetwork.Interface, lb *mgmtnetwork.LoadBalancer, i int, infraID string) bool {
 	id := fmt.Sprintf("%s/backendAddressPools/ssh-%d", *lb.ID, i)
+	ipc := (*nic.InterfacePropertiesFormat.IPConfigurations)[0]
```

Comment:
We also need a check for the IPConfiguration's validity at the beginning of this function.

Reply:
Updated check for NIC IPConfiguration's validity